### PR TITLE
Add GetBitDepth for decoder & use this to set dtype for dlpack

### PR DESCRIPTION
**Pull request by LakshmiKumar23**

### CODE REVIEW

File: src/roc_pybuffer.cpp [hunk 1]
```diff
@@ -149,8 +149,13 @@ int BufferInterface::LoadDLPack(std::vector<size_t>& _shape, std::vector<size_t>
     }
     int itemSizeDT = sizeof(uint8_t); 
     
-    m_dlTensor->dtype.code = kDLUInt;
-    m_dlTensor->dtype.bits = 8;
+    if (bit_depth == 8) {
+        m_dlTensor->dtype.code = kDLUInt;  
+        m_dlTensor->dtype.bits = 8;
+    } else if (bit_depth == 10) {
+        m_dlTensor->dtype.code = kDLInt;  // TODO: add support for 12-bit when required
```
Comment (jeffqjiangNew) [inline src/roc_pybuffer.cpp]: We should use unsigned integer type kDLUInt, same as 8 bit.

File: src/roc_pybuffer.cpp [hunk 2]
```diff
@@ -149,8 +149,13 @@ int BufferInterface::LoadDLPack(std::vector<size_t>& _shape, std::vector<size_t>
     }
     int itemSizeDT = sizeof(uint8_t);
```
Comment (jeffqjiangNew) [inline src/roc_pybuffer.cpp]: This is hard coded and needs review. Lets discuss offline along with the stride changes.

**Review by jeffqjiangNew — CHANGES_REQUESTED** [review]

File: src/roc_pybuffer.cpp (hunk 2 above)
Comment (essamROCm) [inline src/roc_pybuffer.cpp]: There are multiple places in the source code that the change of the uint8 to any size will have an impact, such as when copying the actual frame data based on each element size, 8 or 16 bit, etc. 

Also, the LoadDlPack copying of a frame is currently assume NV12 plane order, must also support YUV444 and P016 plane order.

File: src/roc_pybuffer.cpp (hunk 1 above)
Comment (essamROCm) [reply src/roc_pybuffer.cpp]: Any change in the bit depth, as it has repel-effect in the code, must be tested eventually by saving the tensor in the python example, and visually examine it.

Comment (LakshmiKumar23) [reply src/roc_pybuffer.cpp]: We tested this with a 10bit video and visually looked at it. It looks good. I have handled the bit sizes across all places where it occurs

File: src/roc_pybuffer.cpp (hunk 2 above)
Comment (LakshmiKumar23) [reply src/roc_pybuffer.cpp]: Pushing a change for the itemSize in my next commit and handling stride accordingly

File: src/roc_pybuffer.cpp [hunk 3]
```diff
@@ -60,7 +60,10 @@ py::tuple BufferInterface::strides() const {
 }
 
 std::string BufferInterface::dtype() const {
-    return std::string("|u1");
+    if (m_dlTensor->dtype.code == kDLUInt && m_dlTensor->dtype.bits == 8)
```
Comment (AryanSalmanpour) [inline src/roc_pybuffer.cpp]: remove `m_dlTensor->dtype.code == kDLUInt` as it is the same for both cases.

File: src/roc_pyvideodecode.cpp [hunk 4]
```diff
@@ -164,24 +164,26 @@ int PyRocVideoDecoder::PyDecodeFrame(PyPacketData& packet) {
...
+            stride.push_back(static_cast<size_t>(surf_stride));
+            stride.push_back(sizeof(uint8_t));
         } else if (bit_depth == 10) {
-            stride.push_back(static_cast<size_t>(surf_stride / 2));
+            type_str = static_cast<const char*>("|u2");
+            stride.push_back(static_cast<size_t>(surf_stride));
+            stride.push_back(sizeof(uint16_t));
         }
-        stride.push_back(1);
+        std::vector<size_t> shape{ static_cast<size_t>(height * 1.5), static_cast<size_t>(width)};      //height is multiplied by 1.5 for interleaved YUV
```
Comment (AryanSalmanpour) [inline src/roc_pyvideodecode.cpp]: change the `interleaved YUV` to `NV12 format` in the comment section.

**Review by AryanSalmanpour — CHANGES_REQUESTED** [review]

**Review by jeffqjiangNew — APPROVED** [review]

**Review by AryanSalmanpour — APPROVED** [review]

**Pull request merged**